ports/raspberrypi/common-hal/socketpool/Socket.c

```diff
@@ -680,17 +686,23 @@ bool socketpool_socket(socketpool_socketpool_obj_t *self,
 
     switch (socket->type) {
         case SOCKETPOOL_SOCK_STREAM:
+            MICROPY_PY_LWIP_ENTER;
```

Comment:
another set of calls to check with micropython about.

I compared this to https://github.com/raspberrypi/pico-examples/blob/master/pico_w/wifi/tcp_client/picow_tcp_client.c#L190 and I don't know _WHY_ (i.e., failed at finding specific documentation) but it seems like neither pico-sdk nor micropython guard `tcp_new` .. the pico example does a bunch of calls and then guards only `tcp_connect`, which is very ... specific.

(I won't call out additional sites where you added the enter/exit calls in this file but I spot checked them and they're all absent in micropython from what I could tell)

Reply:
`tcp_new` definitely needs the lock since it calls `LWIP_ASSERT_CORE_LOCKED`. This was one of the first asserts that fired for me in my testing.